### DISCUSSION TITLE
Compare foundry measurements with model performances

### DISCUSSION THREAD

**Discussion by dwarning** [ra3xdh/qucs_s#733] [Q&A]
I use s-parameter files to compare foundry measurements with model performances. Question is, how can I use s-parameter from different operating points in parametric simulations only in plotting. Tried to put them in one file doesn't work (with and without an empty line between the data blocks). Using for each file an extra 2-port setup? Looks a bit elaborate.

_Originally posted by @dwarning in https://github.com/ra3xdh/qucs_s/discussions/475#discussioncomment-8450011_

**Comment by dwarning** [comment]
@ra3xdh : Could you comment please, Thanks.

**Comment by ra3xdh** [comment]
I didn't fully understand what are trying to do. Could you submit a schematic? The s2p implementation for Ngspice is taken froms2spice utiliy: https://groups.io/g/LTspice/topic/75745954?p=Created,,,20,2,0,0 It may have limitations. If the s2p features for Qucs-S+Ngspice are not sufficient you may use Qucs-S+Qucsator. Qucsator has native support of touchstone files.

**Comment by dwarning** [comment]
Thanks.
The problem is not to generate or to convert the s-param files. I have them for different operating points each. But I am not able to show them all in compare to the parametric simulation sweep.

![Bildschirmfoto vom 2024-02-27 18-22-21](https://github.com/ra3xdh/qucs_s/assets/18441932/9835b72c-a8d2-49e2-9eb0-443a6c1c7756)

Only one curve (red) from one file is shown here. The other not. I want only show the data - not simulate with them.

I can provide schematic/project, but you have to include IHP model lib.

**Comment by onehebe** [comment]
Hi dwarning,
As far as I can see, the standard  frequency-based snp file doesn't contains bias/op information.
If multiple blocks of data with different bias/op included in one file, the parser would terminate at the end of first block.
Maybe you can upload your sample snp file. 
We will see how to improve the snp parser function then.

**Comment by dwarning** [comment]
Attached are the 3 sp files as shown before in my picture. I greped them from IHP's mdl file.
[spar-vbe_078_vcb_025.s2p.txt](https://github.com/ra3xdh/qucs_s/files/14535491/spar-vbe_078_vcb_025.s2p.txt)
[spar-vbe_088_vcb_025.s2p.txt](https://github.com/ra3xdh/qucs_s/files/14535492/spar-vbe_088_vcb_025.s2p.txt)
[spar-vbe_096_vcb_025.s2p.txt](https://github.com/ra3xdh/qucs_s/files/14535493/spar-vbe_096_vcb_025.s2p.txt)

**Reply by andresmmera** [reply]
Hi,

I often need to compare different S2P files. To do this in Qucs-S you need to use as many S2P file components as measurements you want to compare. It should look like this:

![Selection_019](https://github.com/ra3xdh/qucs_s/assets/13180689/b5fa5c5c-6142-4c54-a08c-6cd73466942e)
Qucs-S template:  [Plot_Multiple_S2P_prj.zip](https://github.com/user-attachments/files/15537514/Plot_Multiple_S2P_prj.zip)



To speed this up, I have a Javascript tool [1] that allows you to easily upload the files and then compare the traces. Of course, there may be bugs.
![Selection_020](https://github.com/ra3xdh/qucs_s/assets/13180689/3ac132f6-669d-42c0-9fad-bf742d0ba835)

Maybe it would be good to have such a tool in Qucs-S. If @ra3xdh thinks it would be interesting, I volunteered to do it in my spare time.
I think AppCAD [2] (which is free) does almost the same thing.

[1] https://rfdesigntools.pythonanywhere.com/tool/s-parameter-viewer
[2] https://www.broadcom.com/info/wireless/appcad

**Reply by ra3xdh** [reply]
Yes, the S-parameter files viewer would be useful for Qucs-S. @andresmmera , You can start with the porting of this utility.

**Comment by ra3xdh** [comment]
Moved to Discussion/Q&A
